src/Node/ClassStatementsGatherer.php

```diff
@@ -81,6 +88,14 @@ public function getMethods(): array
 		return $this->methods;
 	}
 
+	/**
+	 * @return array<int, int>
+	 */
+	public function getIsTraitMethod(): array
```

Comment:
We can change the return type of `getMethods` to our own object and also add an information about a trait there.

Reply:
with our own object, you mean some new object basically or something different?

Reply:
A new class, ClassStatemenetsGatherer contains many of those. They don't even have to be a `Node`, for example https://github.com/phpstan/phpstan-src/blob/1.7.x/src/Node/Method/MethodCall.php isn't.